### PR TITLE
Slackish dev

### PR DESCRIPTION
**Pull request by slackish**
Ehh, some stuff.  It's not clean, but would appreciate some feedback.

### CODE REVIEW

File: package-puzzles [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+import puzzles
+
+def write_kv_pairs(ziphandle, filename, kv):
+    """ Write out a sorted map to file
+    :param ziphandle: a zipfile object
+    :param filename: The filename to write within the zipfile object
+    :param kv:  the map to write out
+    :return:
+    """
+    filehandle = io.StringIO()
```
Comment (nealey) [inline package-puzzles]: Why the intermediate step of a StringIO instead of writing directly to ziphandle?

File: package-puzzles [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+            for val in kv[key]:
+                filehandle.write("%s: %s%s" % (key, val, os.linesep))
+        else:
+            filehandle.write("%s: %s%s" % (key, kv[key], os.linesep))
+    filehandle.seek(0)
+    ziphandle.writestr(filename, filehandle.read())
+   
+if __name__ == '__main__':        
+    parser = argparse.ArgumentParser(description='Build a category package')
+    parser.add_argument('seed', help='contest seed')
```
Comment (nealey) [inline package-puzzles]: I have a lot of thoughts about this. I'll come talk to you.

File: package-puzzles [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        puzzles_dict = {}
+        secrets = {}
+
+        categoryname = os.path.basename(categorydir.strip(os.sep))
+        
+        # build category seed
+        category_seed = hashlib.new('sha1')
+        category_seed.update(categoryname.encode('utf-8'))
+        category_seed.update(args.seed.encode('utf-8'))
+        category_seed = category_seed.hexdigest()
```
Comment (nealey) [inline package-puzzles]: Need to use the existing category seed if one exists in the zipfile.

File: package-puzzles [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+                    assoc_files.append(fobj.name)
+                zf.writestr(os.path.join(puzzledir, fobj.name), \
+                                    fobj.handle.read())
+
+
+            if len(assoc_files) > 0:
+                puzzlebody.append("")
+                puzzlebody.append("## Associated Files:")
+                for f in assoc_files:
+                    puzzlebody.append(f)
```
Comment (nealey) [inline package-puzzles]: Just want to write down somewhere that we've decided "associated files" needs to go in the json object for the puzzle, not in the body.

File: package-puzzles (hunk 1 above)
Comment (slackish) [inline package-puzzles]: zipfile does not support the ability to write to a file handle directly.  The writing operations supported by ZipFile are to write a string as a file (ZipFile.writestr()) and or to give zipfile a file to write (ZipFile.write()).  I figured StringIO is a suitable intermediate step.

**Comment by nealey** [comment]
We switched branches underneath you. Please re-propose this as a commit against the develop branch. Actually, it's already been merged, so I'm just going to reject this and you still get the log entries.

**Pull request closed without merging**